### PR TITLE
Deployable and tadpole fixes

### LINKED ISSUE #9366: Deployable MG scope bug
### Description

When using the scope on a deployed MG such as the MG-08/495 and another player picks up the MG, you get stuck zoomed out. Reconnecting fixes the visual issue, but you are unable to scope again for the rest of the round. Probably related to this prior bug: https://github.com/tgstation/TerraGov-Marine-Corps/issues/8105

### Test Merges

#9355: 'Debug pr to find out where that damn sleep is' by BraveMole at commit e6dc9a479e
#9333: 'Fix some move issues' by BraveMole at commit 07660a94d2
#9266: 'Pain update (tm first)' by BraveMole at commit e7fc7ada70
#8932: 'Icarus Shipyard. (New map)' by vilereaver41 at commit c49d622ff9
#9239: 'Adds the 6E Chameleon Cape' by hyper2snyper at commit 7148efa111
#9207: 'Add teleporters pad to req' by BraveMole at commit bed7f685a7
#9227: 'Evacuation can prevent xeno major' by BraveMole at commit 1f287382ed

### Reproduction Steps

1. Deploy and use the scope on a machine gun such as the MG-08/495
2. Have another player pick up the machine gun 
3. Your camera is now messed up until you reconnect, and scopes are broken for you until you die or the round ends

### Screenshots

_No response_

### LINKED ISSUE #9513: Spamming launch on tadpole lags the server greatly
### Description

read title

### Test Merges

doesn't matter

### Reproduction Steps

1. Land with tadpole
2. Panic
3. Spam launch
4. Watch the world around you crumble

### Screenshots

https://user-images.githubusercontent.com/24631139/152607545-23575217-fe4e-47c2-bbdc-92ff7ce7594a.mp4

### PR DESCRIPTION
**Pull request by KuzyXD**
## About The Pull Request

fixes #9513 
fixes #9366
when you use deployment in aim mode and want to change direction, you will enter in aim mode automaticly

## Changelog
:cl:
add: aim mode automatically while changing direction using deployable (if you turn on aim mode)
qol: disassemble when someone using deployable is no longer possible
fix: Spamming launch on tadpole no longer lags the server greatly
/:cl:

### CODE REVIEW

File: code/game/objects/machinery/deployable.dm [hunk 1]
```diff
@@ -113,6 +113,10 @@
 	if(CHECK_BITFIELD(internal_item.flags_item, DEPLOYED_WRENCH_DISASSEMBLE))
 		to_chat(user, "<span class = 'notice'>You cannot disassemble [src] without a wrench.</span>")
 		return
+	if(operator)
+		to_chat(user, "<span class = 'notice'>You cannot disassemble [src] when someone uses it.</span>")
```
Comment (hyper2snyper) [inline code/game/objects/machinery/deployable.dm:117]:
```suggestion
		to_chat(user, span_notice("You cannot disassemble [src] while someone is using it."))
```

File: code/modules/projectiles/mounted.dm [hunk 2]
```diff
@@ -173,35 +173,50 @@
...
+	operator.visible_message("[operator] rotates [src].","You rotate [src].")
 	return FALSE
 
+/obj/machinery/deployable/mounted/proc/set_auto_aim_mode(aim_toogle, gun, operator)
+	var/obj/item/weapon/gun/gun_with_new_dir = gun
+
+	if(aim_toogle)
+		gun_with_new_dir.toggle_aim_mode(operator)
+
+/obj/machinery/deployable/mounted/proc/can_be_rotated(angle)
```
Comment (hyper2snyper) [inline code/modules/projectiles/mounted.dm:214]: unless this is called another place, dont make it a new proc.

**Review by hyper2snyper — CHANGES_REQUESTED** [review]
should probably atomize the dropship stuff, but whatever

File: code/modules/projectiles/mounted.dm (hunk 2 above)
Comment (KuzyXD) [reply code/modules/projectiles/mounted.dm:214]: I did this for the sake of code readability.

Comment (KuzyXD) [reply code/modules/projectiles/mounted.dm:214]: I thought about what you said. I'll revert those changes.

**Comment by KuzyXD** [comment]
Is it still need?

**Comment by KuzyXD** [comment]
reopen it please

**Pull request closed without merging**